include/alloc-inl.h

```diff
@@ -673,78 +619,96 @@ static inline size_t next_pow2(size_t in) {
 
 }
 
-/* This function makes sure *size is > size_needed after call.
- It will realloc *buf otherwise.
- *size will grow exponentially as per:
- https://blog.mozilla.org/nnethercote/2014/11/04/please-grow-your-buffers-exponentially/
- Will return NULL and free *buf if size_needed is <1 or realloc failed.
- @return For convenience, this function returns *buf.
- */
-static inline void *maybe_grow(void **buf, size_t *size, size_t size_needed) {
+struct maybe_grow_buf {
```

Comment:
this is now a function with just two variables and nothing else?

Reply:
Aah no this is just the internal struct used by the actual function to calculate offsets in a less.obscure way :)